### DISCUSSION TITLE
What Monaco editor versions are compatible?

### DISCUSSION THREAD

**Discussion by chosen1616** [TypeFox/monaco-languageclient#432] [Q&A] [answered]
In the README troubleshooting, you mention

> If you use monaco-languageclient make sure you have a version of monaco-editor installed in your project that is compliant with monaco-languageclient

Sorry I am new to TypeScript community, I wonder what versions of monaco-editor are compliant? Where can I find this info?

**Comment by kaisalmen** [comment, marked as answer]
Hi @chosen1616 thank you for bringing this up.
The **monaco-editor** version used must be in line with what **monaco-vscode-api** uses.

You can see this here:
```shell
npm list --workspace ./packages/client --all
```
Result
```shell
`-- monaco-languageclient@4.0.1 -> .\packages\client
  +-- vscode-jsonrpc@8.0.2
  +-- vscode-languageclient@8.0.2
  | +-- minimatch@3.1.2
  | | `-- brace-expansion@1.1.11
  | |   +-- balanced-match@1.0.2
  | |   `-- concat-map@0.0.1
  | +-- semver@7.3.8
  | | `-- lru-cache@6.0.0
  | |   `-- yallist@4.0.0
  | `-- vscode-languageserver-protocol@3.17.2
  |   +-- vscode-jsonrpc@8.0.2 deduped
  |   `-- vscode-languageserver-types@3.17.2
  `-- vscode@npm:@codingame/monaco-vscode-api@1.69.12
    +-- monaco-editor@0.34.1
    +-- vscode-oniguruma@1.6.2
    `-- vscode-textmate@7.0.3
```

We should add a table to the README listing the compatible versions and eventually "enforce" the compatibility check with a proper `peerDependencies` in the **package.json** (needs assessment/testing). I created issue #433 to track this.
